File: FiguresLib/Circle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FiguresLib
{
    public class Circle : Figure
    {
        private double _radius;                 // radius of the circle
        public Circle(double radius)
        {
            if (radius >= 0)
                _radius = radius;
            else
                throw new ArgumentException(String.Format("{0} is not a positive number or zero", radius), "radius");
        }
        public double GetRadius() => _radius;
        public override double GetArea() => Math.PI * _radius * _radius;
        public override double GetPerimeter() => 2 * Math.PI * _radius;
    }
}

File: Lab6Cpp/Circle.cpp
#include "Circle.h"

using namespace std;

Circle::Circle(double radius)
{
    if (radius >= 0)
        _radius = radius;
    else
        throw std::invalid_argument("Given radius is not a positive number or zero");
}

double Circle::GetRadius() {
    return _radius;
}

double Circle::GetArea() {
    return M_PI * _radius * _radius;
}

double Circle::GetPerimeter() {
    return 2 * M_PI * _radius;
}